server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/DatabaseChangeRuntimeFlowableTask.java

```diff
@@ -203,26 +219,67 @@ private DatabaseChangeThread generateOdcAsyncTaskThread(Long taskId, DelegateExe
     }
 
     private void modifyTimeoutIfTimeConsumingSqlExists(DelegateExecution execution,
-            DatabaseChangeParameters parameters) {
-        if (!parameters.isModifyTimeoutIfTimeConsumingSqlExists()) {
+            DatabaseChangeParameters parameters, DialectType dialectType, Long creatorId) {
+        long autoModifiedTimeout = flowTaskProperties.getIndexChangeMaxTimeoutMillisecond();
+        if (!parameters.isModifyTimeoutIfTimeConsumingSqlExists() || !dialectType.isOceanbase()
+                || autoModifiedTimeout <= parameters.getTimeoutMillis()) {
             return;
         }
-        Long taskId = FlowTaskUtil.getTaskId(execution);
-        Long preCheckTaskId = FlowTaskUtil.getPreCheckTaskId(execution);
-        TaskEntity preCheckTask = taskService.detail(preCheckTaskId);
-        PreCheckTaskResult preCheckResult =
-                JsonUtils.fromJson(preCheckTask.getResultJson(), PreCheckTaskResult.class);
-        Validate.notNull(preCheckResult, "Pre check task result can not be null");
-        long autoModifiedTimeout = flowTaskProperties.getIndexChangeMaxTimeoutMillisecond();
-        if (Objects.nonNull(preCheckResult.getSqlCheckResult())
-                && preCheckResult.getSqlCheckResult().isTimeConsumingSqlExists()
-                && autoModifiedTimeout > parameters.getTimeoutMillis()) {
-            this.autoModifyTimeout = true;
-            parameters.setTimeoutMillis(autoModifiedTimeout);
-            TaskEntity databaseChangeTaskEntity = taskService.detail(taskId);
-            databaseChangeTaskEntity.setParametersJson(JsonUtils.toJson(parameters));
-            taskService.updateParametersJson(databaseChangeTaskEntity);
+        List<OffsetString> userInputSqls = null;
+        SqlStatementIterator uploadFileSqlIterator = null;
+        String delimiter = parameters.getDelimiter();
+        if (StringUtils.isNotBlank(parameters.getSqlContent())) {
+            userInputSqls = SqlUtils.splitWithOffset(dialectType, parameters.getSqlContent(), delimiter);
```

Comment:
sql content may contains comment, like 'ALTER T ADD  INDEX IDX_1 ; -- some comments', 
spilt result is two sql ['ALTER T ADD  INDEX IDX_1', 'some comments'], second sql is invalid, 
suggest spit with remove comment

Reply:
I can not unstand you mention. The comments will not be a whole SQL even spit without remove comment.

Reply:
comments  will be treat as sql

Reply:
There is really no need to return comment here. 
will use `SqlUtils.splitWithOffset(dialectType, parameters.getSqlContent(), delimiter, true)` method